git_wrapper/repo.py

```diff
@@ -189,3 +189,90 @@ def commit(self, new_commit):
         if not isinstance(new_commit, GitCommit):
             raise TypeError("Commit must be a GitCommit object.")
         self._commit = new_commit
+
+    def sync(self, remote="origin"):
+        """Refresh (fetch) the specified remote
+
+           :param str remote: Remote to sync
+        """
+        try:
+            remote = self.repo.remote(remote)
+        except ValueError:
+            msg = "Remote {remote} does not exist on repo {repo}".format(
+                remote=remote,
+                repo=self.repo.working_dir
+            )
+            raise exceptions.ReferenceNotFoundException(msg)
```

Comment:
I like this check, I am wondering if we should pull this out like reference_exists and make it a decorator. The other idea would be make it a utility that library users could call to check to see if a remote exists in the repo.

Reply:
Cool! I think I would rather wait until it's needed elsewhere before extracting it for now. Maybe we'll understand better if the utility way or the decorator way makes more sense then?

Reply:
That's fair. If we find more need in other methods then we can pull it out then.